autoagora/model_builder.py

```diff
@@ -27,6 +32,63 @@ async def apply_default_model(subgraph: str):
     await set_cost_model(subgraph, model)
 
 
+# (mrq) stands for multi root query
+async def mrq_model_builder(subgraph: str, pgpool: asyncpg.Pool) -> str:
+    logs_db = LogsDB(pgpool)
+    # Obtain most queried mrq
+    most_frequent_multi_root_queries = (
+        await logs_db.get_most_frequent_queries_null_time(subgraph)
+    )
+    for mrq_info in most_frequent_multi_root_queries:
+        await obtain_query_time(subgraph, mrq_info, logs_db)
+    # Call tables with info created
+    most_frequent_queries = await logs_db.get_most_frequent_queries(
+        subgraph, mrq_table=True
+    )
+    model = build_template(subgraph, most_frequent_queries)
+    return model
+
+
+# Obtains the execution time for n amount of random queries
+async def obtain_query_time(
+    subgraph, multi_root_query_info, logs_db, iterations: int = 100
+):
+    # Call db to obtain related variable lists
+    query_log_ids = await logs_db.get_query_logs_id(
+        multi_root_query_info.hash
+    )
+    no_queries = len(query_log_ids)
+    for _ in range(iterations):
+
+        query_id = query_log_ids[random.randint(0, no_queries-1)][0]
+        query_variables = await logs_db.get_query_variables(query_id)
+        query_vars = {f"_{i}": var for i, var in enumerate(query_variables)}
+
+        async with Client(
+            transport=AIOHTTPTransport(args.graph_node_query_endpoint),
+            fetch_schema_from_transport=False,
+        ) as session:
+
+            start_q_execution = time.time()
```

Comment:
I just discovered about this! https://pymotw.com/3/time/#monotonic-clocks
```suggestion
            start_q_execution = time.monotonic()
```